packages/taco/src/conditions/base/shared.ts

```diff
@@ -1,11 +1,41 @@
 import { z } from 'zod';
 
-import { ETH_ADDRESS_REGEXP, USER_ADDRESS_PARAM } from '../const';
+import {
+  CONTEXT_PARAM_REGEXP,
+  ETH_ADDRESS_REGEXP,
+  USER_ADDRESS_PARAM,
+} from '../const';
+import { CONTEXT_PARAM_PREFIX } from '../context/context';
+
+export const contextParamSchema = z.string().regex(CONTEXT_PARAM_REGEXP);
+// We want to discriminate between ContextParams and plain strings
+// If a string starts with `:`, it's a ContextParam
+export const plainStringSchema = z.string().refine(
+  (str) => {
+    return !str.startsWith(CONTEXT_PARAM_PREFIX);
+  },
+  {
+    message: 'String must not be a context parameter i.e. not start with ":"',
+  },
+);
+
+export const paramSchema: z.ZodSchema = z.union([
```

Comment:
Is `paramSchema` ever used externally - wondering if it doesn't need to be exported.